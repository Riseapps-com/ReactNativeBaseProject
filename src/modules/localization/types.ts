export type TranslationGetters = {
  [key: string]: () => Record<string, any>;
};
