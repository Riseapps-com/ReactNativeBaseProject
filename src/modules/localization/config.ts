export const LOCALIZATION_FALLBACK_LANGUAGE = 'en';
