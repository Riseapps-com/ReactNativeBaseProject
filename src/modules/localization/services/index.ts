export { default as I18n } from './I18n';
