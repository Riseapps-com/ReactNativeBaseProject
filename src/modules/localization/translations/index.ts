import { TranslationGetters } from '../types';
import de from './de.json';
import en from './en.json';

const translations: TranslationGetters = {
  en: () => en,
  de: () => de,
};

export { translations };
