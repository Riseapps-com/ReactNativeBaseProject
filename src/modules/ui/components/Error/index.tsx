import React from 'react';
import { useTranslation } from 'react-i18next';
import { View } from 'react-native';

import { testIDs } from '~config';
import { useTheme } from '~theme';

import Text from '../Text';
import themedStyles from './styles';

const Error: React.FC = () => {
  const [styles] = useTheme(themedStyles);
  const { t } = useTranslation();

  return (
    <View testID={testIDs.global.error} style={styles.wrapper}>
      <Text fontStyle={'bold'} size={'bigger'} style={styles.error}>
        {t('errorBoundaryTitle')}
      </Text>
    </View>
  );
};

export default Error;
