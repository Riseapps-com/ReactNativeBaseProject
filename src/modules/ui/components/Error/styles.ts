import { styleSheetFactory } from '~theme';

export default styleSheetFactory(theme => ({
  wrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },

  error: {
    color: theme.error,
  },
}));
