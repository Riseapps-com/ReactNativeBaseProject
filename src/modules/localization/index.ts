import * as localizationConfig from './config';

export * from './services';

export { localizationConfig };
