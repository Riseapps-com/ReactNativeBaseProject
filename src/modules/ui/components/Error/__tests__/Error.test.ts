import { I18n } from '~modules/localization';
import { renderComponent } from '~modules/tests';
import { colors } from '~theme';

import Error from '../index';

const renderError = () => renderComponent(Error);

describe('ui', () => {
  describe('<Error />', () => {
    it('renders error correctly', () => {
      const error = renderError();

      expect(error).toMatchSnapshot();
    });

    it('renders error with correct title and style', () => {
      const error = renderError();

      expect(error.getByText(I18n.t('errorBoundaryTitle'))).toBeTruthy();
      expect(error.getByText(I18n.t('errorBoundaryTitle'))).toHaveStyle({
        color: colors.SUNSET_ORANGE,
      });
    });
  });
});
