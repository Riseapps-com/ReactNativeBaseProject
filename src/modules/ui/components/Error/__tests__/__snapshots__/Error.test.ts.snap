// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui <Error /> renders <Error /> 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="global_error"
>
  <Text
    style={
      Array [
        Object {
          "color": "#4A4E4D",
        },
        Object {
          "fontFamily": "Quicksand-Bold",
        },
        Object {
          "fontSize": 18,
        },
        Object {
          "color": "#fe4a49",
        },
      ]
    }
  >
    errorBoundaryTitle
  </Text>
</View>
`;
